variant_concordance/var_conc_vcf.wdl

```diff
@@ -0,0 +1,223 @@
+
+        # convert vcfs to bed keeping wanted samples
+        plink2 --allow-extra-chr --vcf ${vcf1} --keep keep_vcf1_samples --freq --make-bed --out ${base1}
+        plink2 --allow-extra-chr --vcf ${vcf2} --keep keep_vcf2_samples --freq --make-bed --out ${base2}
+
+        # filter beds so they will have the same variants
+        plink2 --allow-extra-chr --extract <(cut -f2 ${base2}.bim) --bfile ${base1} --make-bed --out ${base1}_shared
+        plink2 --allow-extra-chr --extract <(cut -f2 ${base1}.bim) --bfile ${base2} --make-bed --out ${base2}_shared
+
+        # underscores cause trouble with vcf conversions
+        sed -i 's/_/-/g' ${base1}_shared.fam
+        sed -i 's/_/-/g' ${base2}_shared.fam
+
+        # in case sample names are shared between datasets, add suffix to one dataset's samples
+        awk 'BEGIN{OFS="\t"} {$2=$2"-DATA1"} 1' ${base1}_shared.fam > temp && mv temp ${base1}_shared.fam
+        awk 'BEGIN{OFS="\t"} {$1=$1"-DATA1"} 1' ${sample_pairs} > sample_pairs_suffixed
+
+        # merge datasets
+        plink --keep-allele-order --allow-extra-chr --bfile ${base1}_shared --bmerge ${base2}_shared --make-bed --out merged
+
+        # convert to vcf
+        plink2 --bfile merged --recode vcf-iid bgz --output-chr chrM --out ${outname}
+        #tabix -p vcf ${outname}.vcf.gz
+
+    >>>
+
+    runtime {
+        cpu: 4
+        memory: "3G"
+        disks: "local-disk 200 HDD"
+        docker: "eu.gcr.io/finngen-refinery-dev/bioinformatics:0.7"
+        zones: "europe-west1-b europe-west1-c europe-west1-d"
+        preemptible: 1
+    }
+
+    output {
+        File out_vcf = outname + ".vcf.gz"
+        File out_sample_pairs = "sample_pairs_suffixed"
+    }
+}
+
+task concordance {
+
+    File vcf
+    File sample_pairs
+
+    command <<<
+
+        set -euxo pipefail
+
+        # underscores were converted to dashes in the filter_merge task, do the same here to match samples right
+        sed -i 's/_/-/g' ${sample_pairs}
+
+        python3 <<EOF | bgzip > concordance.gz
+
+        """
+        computes variant-wise concordance stats
+        inputs: a vcf file and a text file with a list of corresponding samples one pair of samples on a row
+        prints stats to stdout and messsages to stderr
+        """
+
+        import gzip
+        import sys
+
+        def eprint(*args, **kwargs):
+            print(*args, file=sys.stderr, **kwargs)
+
+        def read_sample_index(vcf):
+            with gzip.open(vcf, 'rt') as f:
+                line = f.readline().strip()
+                while line.startswith('##'):
+                    line = f.readline().strip()
+                if not line.startswith('#CHROM'):
+                    eprint('weird vcf, quit')
+                    # sorry
+                    quit(2)
+                return {sample: i for i,sample in enumerate(line.split('\t'))}
+
+        def count_alt_alleles(gt):
+            if gt == './.':
+                return -1
+            s = gt.split('/')
```

Comment:
gonna be used for phased data fo sho sooner than later so why not split with also |

Reply:
Totally